docs/ADR-65-urn-prefix-search.md

```diff
@@ -0,0 +1,104 @@
+
+# URN prefix search
+
+## Context and Problem Statement
+
+As part of the Third Party Project, the need of exposing all items of a collection appeared. The solution proposed solves a more generic approach of the problem: the ability to list all entities in Content Server which their pointers match the prefix given.
+
+## Considered options
+
+- Expose the filter in Lambdas Server
+- Expose the filter in Content Server
+
+The decision is to solve this at Content Server Level as the approach of the filter can be done directly at database level.
+
+## Decision
+
+The creation of a [new endpoint in the Content Server](https://github.com/decentraland/catalyst-api-specs/pull/21):
+
+`GET /entities/currently-pointed/{urnPrefix}`
+
+That returns a list with all active entityIds that point to at least one pointer with the urnPrefix.
+
+Issue: [@catalyst#878](https://github.com/decentraland/catalyst/issues/878)
+
+
+### Content Server Technical Solution
+Created a new table `active_pointers` to store the information of the active entities per pointer (instead of using the pointers field in the database). This new table has as a key the pointer and the other column is the entityId.
+Modified the deployments of entities to update the row of the modified pointer to store the latest version. Also, removes the row of the table when a deleter deployment is done.
+
+## Status
+
+Accepted
+
+## Consequences
+
+### Don't accept wearables deployments with more than one pointer (URN)
+
+Catalyst Issue: [@catalyst#884](https://github.com/decentraland/catalyst/issues/884)
+
+To support the sync of Base and L1 wearables the check of resolving to the same URN was added, that's okay for those, but it's not accurate for L2 (Polygon) wearables. So, for new wearable deployments exactly one pointer is allowed.
```

Comment:
I don't consider this is a blocker or a need. In the worst case scenario, entities with many pointers will appear in two different queries for different prefixes. The generalization of the problem is to look for entities with prefix `-` that will render a response of around 45k LAND starting with a negative number. Many of those will be repeated, because scenes share many lands

Reply:
For the Platform team, it's not a blocker, as long as the client implements the logic to query an item twice, as it's not the Content Server's responsibility to implement the logic to retrieve both entities when matched.

What do you think @nachomazzara?

Reply:
I think the query must dedup the entities pointed twice. In SQL that should be inverting an inner join order and adding a distinct.

```sql
-- from this
SELECT 
  d.* 
FROM active_pointer a
INNER JOIN deployments d  ON a.pointer IN d.pointers -- pseudo SQL
WHERE ...
```

```sql
-- to this
SELECT DISTINCT
  d.* 
FROM deployments d 
INNER JOIN active_pointer a ON a.pointer IN d.pointers -- pseudo SQL
WHERE ...
```

Reply:
If the deployment in the deployments table contains two pointers, then in the active_pointer there will be two rows. Is that enough?

Reply:
Yes, but the endpoint should return only one entry with two pointers in that case. Are we in the same page?

Reply:
The endpoint response format is:

```
[
{
  "pointer": "onePointer",
  "entityId": "hash"
}
]
```

So, it won't retrieve both pointers. It's client's side responsability